app/components/ui/button.tsx

```diff
@@ -17,53 +17,99 @@ type BaseButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
   let baseClasses =
-    'w-full sm:w-fit me-2 mb-2 text-center font-medium focus:outline-none focus:ring-4 px-3 py-2 text-sm sm:px-5 sm:py-2.5 md:py-3 md:text-base';
-  let variantClasses = '';
-  let styleClasses = '';
-
-  if (variant === 'primary') {
-    variantClasses = 'bg-primary text-white focus:ring-gray-300 hover:bg-gray-900';
-  } else if (variant === 'secondary') {
-    variantClasses = 'bg-secondary focus:ring-teal-300 hover:bg-teal-200';
-  } else if (variant === 'warm') {
-    variantClasses = 'bg-warm text-white focus:ring-red-300 hover:bg-red-700';
-  } else {
-    variantClasses = 'bg-white text-primary border border-gray-200 focus:ring-gray-300 hover:bg-gray-100';
+    'w-full sm:w-fit me-2 mb-2 px-3 sm:px-5 py-2 sm:py-2.5 md:py-3 text-center text-sm md:text-base font-medium focus:outline-none focus:ring-4';
+
+  let variantClasses = getVariantClasses(variant);
+  let styleClasses = getStyleClasses(buttonStyle);
+
+  if (icon) {
+    baseClasses += ' inline-flex items-center justify-center';
   }
 
-  if (buttonStyle === 'rounded') {
-    styleClasses = 'rounded-lg';
-  } else {
-    styleClasses = 'rounded-full';
+  if (disabled) {
+    variantClasses =
+      'w-full sm:w-fit me-2 mb-2 px-3 sm:px-5 py-2 sm:py-2.5 md:py-3 text-center text-sm md:text-base font-medium bg-gray-300 text-gray-600 opacity-75 cursor-not-allowed';
+    baseClasses = '';
   }
 
+  return (
+    <button className={`${baseClasses} ${variantClasses} ${styleClasses}`} {...props}>
+      {children}
+    </button>
+  );
+}
+
+export function AuthButton({ variant, buttonStyle, icon, disabled, children, ...props }: BaseButtonProps) {
+  let baseClasses = 'w-full py-3 text-center text-base font-normal focus:outline-none focus:ring-4 py-3';
+  let variantClasses = getVariantClasses(variant);
+  let styleClasses = getStyleClasses(buttonStyle);
+
   if (icon) {
     baseClasses += ' inline-flex items-center justify-center';
   }
 
   if (disabled) {
-    variantClasses += ' opacity-75 cursor-not-allowed';
+    variantClasses =
+      'w-full py-3 text-center text-base font-normal bg-gray-300 text-gray-600 opacity-75 cursor-not-allowed';
+    baseClasses = '';
   }
 
   return (
-    <button className={`${baseClasses} ${variantClasses} ${styleClasses}`} {...props}>
+    <button type="submit" className={`${baseClasses} ${variantClasses} ${styleClasses}`} {...props}>
+      {children}
+    </button>
+  );
+}
+
+type IconButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
+  'aria-label': string;
+  children: React.ReactNode;
+};
+
+export function IconButton({ 'aria-label': ariaLabel, children, ...props }: IconButtonProps) {
+  return (
+    <button aria-label={ariaLabel} {...props}>
       {children}
     </button>
   );
 }
+
+type LinkButtonProps = {
+  children: ReactNode;
+  to: string;
+};
+
+export function LinkButton({ to, children, ...props }: LinkButtonProps) {
```

Comment:
There exists a pattern where a link is styled as a button (even though it's just an anchor tag). These components are called `LinkButton` or `ButtonLink` in design systems. However, here we just render a link that also looks like a link. I believe we could just rename the component to `Link` (using Remix's `Link` under the hood). In this case, we can also move it out of the button file into a new links file!

Reply:
Oh, sorry! I was confused a bit. Thanks for clarifying. I see what you meant 🙂 

In Material Design, they call them Text Buttons. In this case, we should indeed move it to links!